applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -29,7 +29,28 @@ void BaseSolidElement::Initialize()
 {
     KRATOS_TRY
 
-    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(  );
+    if( GetProperties().Has(INTEGRATION_ORDER) ) {
+        const SizeType integration_order = GetProperties()[INTEGRATION_ORDER];
+        switch ( integration_order )
+        {
+        case 1:
```

Comment:
@jcotela I am already using the enums GeometryData::GI_GAUSS_X

Reply:
What I meant is that, for example, if you know you want GI_GAUSS_1 for linear triangles and GI_GAUSS_2 for quadratic ones (for example), you can select your integration order based on the return of `Geomtery::GetGeometryType()` (which is also an enum) instead of relying on the user.

Reply:
That was the previous implementation (GetDefaultIntegration does that), but you may need for whatever reason to use a higher number of GP, for example for a  specific CL, th problem is that it relies in the user to know that